src/npctalk_rules.cpp

```diff
@@ -0,0 +1,518 @@
+                ImGui::PushID( button_number );
+                button_number++;
+                if( role_model.rules.flags == guy->rules.flags &&
+                    role_model.rules.aim == guy->rules.aim &&
+                    role_model.rules.engagement == guy->rules.engagement &&
+                    role_model.rules.cbm_recharge == guy->rules.cbm_recharge &&
+                    role_model.rules.cbm_reserve == guy->rules.cbm_reserve &&
+                    role_model.rules.pickup_whitelist->empty() == guy->rules.pickup_whitelist->empty() ) {
+                    ImGui::PushStyleColor( ImGuiCol_Button, c_green );
+                }
+                if( ImGui::Button( toggle_label.c_str() ) ) {
+                    do_flags_overrides = true;
+                    do_aim = true;
+                    do_engagement = true;
+                    do_cbm_recharge = true;
+                    do_cbm_reserve = true;
+                    do_pickup = true;
+                }
+                ImGui::PopID();
+                ImGui::PopStyleColor();
+                /*The great swap*/
+                // To make this slightly easier...
+                npc *guy_with_same = exporting_rules ? guy : &role_model;
+                npc *guy_with_new = exporting_rules ? &role_model : guy;
+                if( do_flags_overrides ) {
+                    guy_with_new->rules.flags = guy_with_same->rules.flags;
+                    guy_with_new->rules.overrides = guy_with_same->rules.overrides;
+                    guy_with_new->rules.override_enable = guy_with_same->rules.override_enable;
+                }
+                if( do_aim ) {
+                    guy_with_new->rules.aim = guy_with_same->rules.aim;
+                }
+                if( do_engagement ) {
+                    guy_with_new->rules.engagement = guy_with_same->rules.engagement;
+                }
+                if( do_cbm_recharge ) {
+                    guy_with_new->rules.cbm_recharge = guy_with_same->rules.cbm_recharge;
+                }
+                if( do_cbm_reserve ) {
+                    guy_with_new->rules.cbm_reserve = guy_with_same->rules.cbm_reserve;
+                }
+                if( do_pickup ) {
+                    guy_with_new->rules.pickup_whitelist = guy_with_same->rules.pickup_whitelist;
+                }
+            }
+        }
+        ImGui::EndTable();
+    }
+}
+
+void follower_rules_ui_impl::draw_controls()
+{
+    if( !guy ) {
+        debugmsg( "Something has gone very wrong, can't find NPC to set rules for.  Aborting." );
+        last_action = "QUIT";
+        return;
+    }
+
+    ImGui::SetWindowSize( ImVec2( window_width, window_height ), ImGuiCond_Once );
+
+    static bool exporting_rules = false;
+    static bool in_popup = false;
+    if( in_popup ) {
+        rules_transfer_popup( exporting_rules, in_popup );
+        return;
+    }
+
+    ImGui::InvisibleButton( "TOP_OF_WINDOW_KB_SCROLL_SELECTABLE", ImVec2() );
```

Comment:
This and the bottom of the window one are for scrolling purposes. Otherwise you could scroll all the way down and your nav ends up... above the last line of text, so you cannot see what the last rule is without specifically switching to pageup/pagedown or moving the scrollbar with the mouse.

Reply:
To illustrate: 

https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/28569666-b5ce-4d9b-b1d4-f3d56e3597a2